### ISSUE TITLE
Proposal: Stop reading response body into memory

### ISSUE THREAD

**Issue by krisiasty** [tsenart/vegeta#408]
#### Proposal

Instead of reading part (or all) of the response body into memory, just read and discard the whole body from network. Also, the Bytes In in the report should count the data read from the network (the size of the whole response body).

#### Background

It's not clear to me what is the rationale for reading the response body into memory (limited by max-body parameter), when in the next step the rest of the body is read from network and discarded anyway. The content of the body isn't used anyway, so why waste memory?

The way it works now causes large memory consumption when targets points to the large objects (think s3-compatible object store with several hundreds megabyte or larger objects). 
On the other way, using max-body parameter makes the reported Bytes In metric useless (it doesn't report data read from network, only portion which was read into memory).

#### Workarounds

none

**Comment by sanxun0325** [comment]
I think should keep response body because I want to use the return of the previous interface to continue the attack on the next interface.

**Comment by brennantaylor** [comment]
It also ensures you drain the data out of the stream instead of causing an early close which could give you false confidence of your targets being able to handle the load.

**Comment by krisiasty** [comment]
Doesn't io.Copy(ioutil.Discard, r.Body) read all the content from the target? The difference is that it doesn't allocate unnecessary buffers.
The only place I saw where the Result.Body is really used is in method Result.Equal, which in turn is only used in tests. 

My original concern was that when I tried to use vegeta to test really fast object storage holding thousands of large (>10M) objects, memory allocation and usage became a huge problem, but I just realized that using maxBody=0 option should effectively do what I need.
